src/peft/peft_model.py

```diff
@@ -650,13 +651,42 @@ def create_or_update_model_card(self, output_dir: str):
         Updates or create model card to include information about peft:
         1. Adds `peft` library tag
         2. Adds peft version
-        3. Adds quantization information if it was used
+        3. Adds base model info
+        4. Adds quantization information if it was used
         """
-        # Adds `peft` library tag
-        add_library_to_model_card(output_dir)
 
-        with open(os.path.join(output_dir, "README.md"), "r") as f:
-            lines = f.readlines()
+        filename = os.path.join(output_dir, "README.md")
+        card_exists = True
+        if not os.path.exists(filename):
+            card_exists = False
+            # touch an empty model card
+            with open(filename, "a"):
+                pass
+
+        with open(filename, "r") as f:
+            readme = f.read()
+
+        # add metadata
+        match = re.search(r"---\n(.*?)\n---", readme, re.DOTALL)
+        metainfo = {} if match is None else yaml.safe_load(match.group(1))
+        metainfo["library_name"] = "peft"
+        model_config = self.config
+        if hasattr(model_config, "to_dict"):
+            model_config = model_config.to_dict()
+        if model_config["model_type"] != "custom":
+            metainfo["base_model"] = model_config["_name_or_path"]
+
+        card_data = ModelCardData(**metainfo)
+
+        # add extra data to model card body
+        card = ModelCard.from_template(card_data)
+
+        # check if there is already a text body on the README.md, if not, use default template
+        if card_exists:
+            text = readme.split("\n---\n", 1)[-1]
+        else:
+            text = card.text
+        lines = text.splitlines()
```

Comment:
Here I would use `ModelCard.load(filename)` instead of parsing the existing model card manually. In general it's safer as we iterated several times on `huggingface_hub`'s regex to parse metadata to get a correct one (Windows newlines being especially tricky and errors hard to notice). Also you wouldn't have to process differently depending on if the model card already exists or not: 

```suggestion
        filename = os.path.join(output_dir, "README.md")
        
        card = (
            ModelCard.load(filename)
            if os.path.exists(filename)
            else ModelCard.from_template(ModelCardData())
        )

        card.data["library_name"] = "peft"
        model_config = self.config
        if hasattr(model_config, "to_dict"):
            model_config = model_config.to_dict()
        if model_config["model_type"] != "custom":
            card.data["base_model"] = model_config["_name_or_path"]

        lines = card.text.splitlines()
```

Reply:
Arf, sorry for the huge diff. Hopefully it's still readable :grimacing: